modules/javafx.controls/src/main/java/com/sun/javafx/scene/control/behavior/ButtonBehavior.java

```diff
@@ -178,8 +178,7 @@ protected void keyReleased(KeyEvent e) {
      * potentially arming the Button, this will transfer focus to the button
      */
     protected void mousePressed(MouseEvent e) {
-        // if the button is not already focused, then request the focus
-        if (! getNode().isFocused() && getNode().isFocusTraversable()) {
+        if (getNode().isFocusTraversable()) {
```

Comment:
that should read:

The old code used to skip the call to requestFocus() if the node was already focused (here and in a couple other cases). _It now always calls it_ in order to correctly update the focusVisible state.

Reply:
I don't think there are any side effects, since the old code just returned from `Scene.(KeyHandler).requestFocus(Node)` without doing anything when the node was already the focus owner of the scene.